### PR TITLE
Escape id parameter for queues view

### LINKED ISSUE #1679: Vulnerable to Reflected Cross Site Scripting through pathname
`https://example.com/queues/`

image

As shown in the image above, the application is vulnerable to Reflected Cross Site Scripting through pathname of the site. With Non-Persistent cross-site scripting, malicious code is executed by the victim’s browser, and the payload is not stored anywhere; instead, it is returned as part of the response HTML that the server sends. The vulnerability is typically a result of incoming requests not being sufficiently validated, which allows for the manipulation of a web application’s function and the activation of malicious scripts. In order to exploit this vulnerability, an attacker needs to send the link to the victim. When the victim loads the link, the cross-site scripting payload will execute.

**Recommended fix:**
 HTML ESCAPE: Ruby on Rails has a built-in XSS protection mechanism which automatically HTMLescapes all the data being transferred from Rails to HTML. Rails uses ERB::Util#html_escape function toescape HTML entities. This particular function does the substitution based on the following hash (moreabout ERB::Util). ``` { '&' => '&amp;', '>' => '&gt;', '<' => '&lt;', '"' => '&quot;', "'" => '&#39;' } ``` Here is anexample: ``` <b>Hello <%= params[:name] %></b> ``` If the above is attacked with ?name=<script>alert(1)</script>, the output will be as follows: ``` <b> Hello &lt;script&gt;alert(1)&lt;/script&gt;gt; </b> `

### PR DESCRIPTION
**Pull request by brianvans**
Fixes #1679

### CODE REVIEW

File: lib/resque/server/views/queues.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 <% if current_queue = params[:id] %>
 
-  <h1>Pending jobs on <span class='hl'><%= current_queue %></span></h1>
+  <h1>Pending jobs on <span class='hl'><%= h current_queue %></span></h1>
```
Comment (iloveitaly) [inline lib/resque/server/views/queues.erb:5]: `h` is only available in rails and resque does not depend on rails. This should do the trick:

```suggestion
  <h1>Pending jobs on <span class='hl'><%= ERB::Util.html_escape current_queue %></span></h1>
```

@brianvans does this look good to you?

**Comment by iloveitaly** [comment]
@brianvans what do you think of the suggested change? If you commit/accept, I can work on getting this merged.

File: lib/resque/server/views/queues.erb (hunk 1 above)
Comment (brianvans) [reply lib/resque/server/views/queues.erb:5]: `h` is aliased as a helper over here:

https://github.com/brianvans/resque/blob/70f14b12d495ac0727cfe69fbfa2209fe2c42a31/lib/resque/server.rb#L28-L30

I noticed it was used in a few other views, so it made sense to use it here as well.

Comment (iloveitaly) [reply lib/resque/server/views/queues.erb:5]: Ah, nice! Didn't realize that.

**Review by iloveitaly — APPROVED** [review]

**Comment by iloveitaly** [comment]
@chrisccerami Here's another one ready for merging!

**Comment by iloveitaly** [comment]
@brianvans can you rebase on master so we can get the tests to pass?

**Comment by brianvans** [comment]
> @brianvans can you rebase on master so we can get the tests to pass?

Done!

**Review by iloveitaly — APPROVED** [review]
@chrisccerami could you merge this one in?

**Comment by G-Rath** [comment]
@brianvans @iloveitaly could we get a security advisory published for this?

**Comment by G-Rath** [comment]
cc @corincerami just realised you're the one that did the merging

**Pull request merged**